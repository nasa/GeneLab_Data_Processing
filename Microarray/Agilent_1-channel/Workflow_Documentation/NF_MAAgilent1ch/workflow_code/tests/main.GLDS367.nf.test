nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Direct runsheet version: Should run without failures") {

        when {
            params {
                // define parameters here. Example: 
                runsheetPath = "../examples/runsheet/GLDS-367_microarray_v0_runsheet.csv"
                gldsAccession = "GLDS-367"
                limit_biomart_query = 300
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                            path("${params.outputDir}/${params.gldsAccession}/Metadata").list(),
                            path("${params.outputDir}/${params.gldsAccession}/00-DE").list(),
                            ).match()
        }

    }
    test("GLDS-367: Should run without failures") {

        tag "core"

        when {
            params {
                // define parameters here. Example: 
                osdAccession = "OSD-367"
                gldsAccession = "GLDS-367"
                biomart_attribute = "Agilent wholegenome 4x44k v1"
                limit_biomart_query = 300
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                            path("${params.outputDir}/${params.gldsAccession}/Metadata").list(),
                            path("${params.outputDir}/${params.gldsAccession}/00-RawData").list(),
                            path("${params.outputDir}/${params.gldsAccession}/01-limma_NormExp").list(),
                            path("${params.outputDir}/${params.gldsAccession}/02-limma_DGE").list(),
                            ).match()
        }

    }

}
