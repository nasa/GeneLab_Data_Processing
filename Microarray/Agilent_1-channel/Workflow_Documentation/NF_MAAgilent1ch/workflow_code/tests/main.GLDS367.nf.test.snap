{
    "GLDS-367: Should run without failures": {
        "content": [
            [
                "GLDS-367_metadata_GLDS-367-ISA.zip:md5,130bb4f9fded2750631cfed98c173c06",
                "OSD-367_microarray_v0_runsheet.csv:md5,20d3407f692abaf4471151a7d85c0420"
            ],
            [
                "raw_intensities.csv:md5,21b8fe8485e30baabd33d4341be08a78"
            ],
            [
                "normalized_expression.csv:md5,baae31c588f25cf574df54ee4937cbaa"
            ],
            [
                "SampleTable.csv:md5,d6f57c657e6594a1f52ff9a996118083",
                "contrasts.csv:md5,c9af9056cef7bd0210a626cd9c6175b2",
                "differential_expression.csv:md5,1bce5936134e25585716ebe983743c32",
                "visualization_PCA_table.csv:md5,6b37103ca8cacdfc2a924631e7422fa9"
            ]
        ],
        "timestamp": "2023-04-27T23:01:03+0000"
    },
    "Direct runsheet version: Should run without failures": {
        "content": [
            [
                "GLDS-367_microarray_v0_runsheet.csv:md5,6b066905180d502a70e4ad9fc094433d"
            ],
            [
                "SampleTable.csv:md5,f393473808e2b9e262c7da5809e0baca",
                "contrasts.csv:md5,c9af9056cef7bd0210a626cd9c6175b2",
                "differential_expression.csv:md5,1bce5936134e25585716ebe983743c32",
                "normalized_expression.csv:md5,baae31c588f25cf574df54ee4937cbaa",
                "raw_intensities.csv:md5,21b8fe8485e30baabd33d4341be08a78",
                "visualization_PCA_table.csv:md5,6b37103ca8cacdfc2a924631e7422fa9",
                "visualization_output_table.csv:md5,e1aa48627dde500d26443891de6c23d9"
            ],
            {
                "Agile1CMP.html exists": true
            }
        ],
        "timestamp": "2023-04-27T23:01:02+0000"
    }
}